scripts/ingests/2020Best_source_ingest.py

```diff
@@ -0,0 +1,78 @@
+import sys
+from sqlalchemy.sql.elements import Null
+sys.path.append('.')
+from astrodbkit2.astrodb import create_database
+from astrodbkit2.astrodb import Database
+from simple.schema import *
+from utils import *
+from astropy.table import Table
+from pathlib import Path
+import os
+
+DRY_RUN = False #modifies .db file but not the data files
+RECREATE_DB = True #recreates the .db file from the data files
+VERBOSE = False
+
+verboseprint = print if VERBOSE else lambda *a, **k: None
+
+db_file = 'SIMPLE.db'
+db_file_path = Path(db_file)
+db_connection_string = 'sqlite:///SIMPLE.db'  # SQLite browser
+
+if RECREATE_DB and db_file_path.exists():
+    os.remove(db_file) #removes the current .db file if one already exists
+
+if not db_file_path.exists():
+    create_database(db_connection_string) #creates empty database based on the simple schema
+    db = Database(db_connection_string) #connects to the empty database
+    db.load_database('data/') #loads the data from the data files into the database
+else:
+    db = Database(db_connection_string) #if database already exists, connects to .db file
+
+# load table of sources to ingest
+Best = Table.read('UltracoolSheet-Main.csv')
+
+# find sources already in database
+existing_sources = []
+missing_sources = []
+db_names = []
+for i,name in enumerate(Ydwarfs['source']):
+	if len(db.search_object(name,resolve_simbad=True)) != 0:
+		existing_sources.append(i)
+		db_names.append(db.search_object(name,resolve_simbad=True)[0].source)
+	else:
+		missing_sources.append(i)
+		db_names.append(Ydwarfs['source'][i])
+
+# add missing references
+ref_list = Ydwarfs['reference'].tolist()
+included_ref = db.query(db.Publications.c.name).filter(db.Publications.c.name.in_(ref_list)).all()
+included_ref = [s[0] for s in included_ref]
+new_ref = list(set(ref_list)-set(included_ref))
+new_ref = [{'name': s} for s in new_ref]
+
+if len(new_ref)>0:
+	db.Publications.insert().execute(new_ref)
```

Comment:
Get rid of this...this is now taken care of by `search_publication` and `add_publication` functions.

Reply:
So should I instead set up a function where it uses search_publication to find the missing publications and then use add_publication to add them?

Reply:
doesn't even have to be another function.
```
if search_publication(ref) == False:
    add_publication(ref)

Reply:
The issue is, when I used the add_publication function, it says it needs a name and bibcode to be entered